### DISCUSSION TITLE
如何确定当前target的语言？

### DISCUSSION THREAD

**Discussion by Extra-Creativity** [xmake-io/xmake#4900] [Q&A]
比如项目里同时有cuda和cpp，想写一个rule只对cpp的部分`add_defines`。

**Comment by star-hengxing** [comment]
脚本域获取所有 files，判断如果是 cpp 后缀就重新 add files 把宏附带进去

**Reply by Extra-Creativity** [reply]
感觉xmake可以开放接口，毕竟内部肯定也是要判断是什么语言的...

比如CMake有generator expression

**Reply by star-hengxing** [reply]
https://xmake.io/#/manual/target_instance?id=targetsourcebatches

**Reply by waruqi** [reply]
直接用 add_cxxflags 加不就好了么

**Reply by Extra-Creativity** [reply]
`add_cxflags`只对特定编译器能用吧，msvc和gcc定义宏的选项不一样（`/D` or `-D`），应该没法像`add_defines`一样统一吧。

**Reply by star-hengxing** [reply]
msvc 的 `/` 和 `-` 一样

**Comment by waruqi** [comment]
> `add_cxflags`只对特定编译器能用吧，msvc和gcc定义宏的选项不一样（`/D` or `-D`），应该没法像`add_defines`一样统一吧。

`-Dxx` 目前基本跨所有 c++ 编译器。。至少我没见过不支持的。。
